x-pack/plugins/ml/public/embeddables/single_metric_viewer/single_metric_viewer_initializer.tsx

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useState } from 'react';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiForm,
+  EuiFormRow,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiFieldText,
+  EuiModal,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { MlJob } from '@elastic/elasticsearch/lib/api/types';
+import type { SingleMetricViewerServices } from '..';
+import { TimeRangeBounds } from '../../application/util/time_buckets';
+import { SeriesControls } from '../../application/timeseriesexplorer/components/series_controls';
+import {
+  APP_STATE_ACTION,
+  type TimeseriesexplorerActionType,
+} from '../../application/timeseriesexplorer/timeseriesexplorer_constants';
+
+export interface SingleMetricViewerInitializerProps {
+  bounds: TimeRangeBounds;
+  defaultTitle: string;
+  initialInput?: SingleMetricViewerServices;
+  job: MlJob;
+  onCreate: (props: {
+    panelTitle: string;
+    functionDescription: string;
+    selectedDetectorIndex: number;
+    selectedEntities: any;
+  }) => void;
+  onCancel: () => void;
+}
+
+export const SingleMetricViewerInitializer: FC<SingleMetricViewerInitializerProps> = ({
+  bounds,
+  defaultTitle,
+  initialInput,
+  job,
+  onCreate,
+  onCancel,
+}) => {
+  const [panelTitle, setPanelTitle] = useState<string>(defaultTitle);
+  const [functionDescription, setFunctionDescription] = useState<string>('');
+  const [selectedDetectorIndex, setSelectedDetectorIndex] = useState<number>(0);
+  const [selectedEntities, setSelectedEntities] = useState<any>();
+
+  const isPanelTitleValid = panelTitle.length > 0;
+
+  const handleStateUpdate = (action: TimeseriesexplorerActionType, payload: any) => {
+    switch (action) {
+      case APP_STATE_ACTION.SET_ENTITIES:
+        setSelectedEntities(payload);
+        break;
+      case APP_STATE_ACTION.SET_FUNCTION_DESCRIPTION:
+        setFunctionDescription(payload);
+        break;
+      case APP_STATE_ACTION.SET_DETECTOR_INDEX:
+        setSelectedDetectorIndex(payload);
+        break;
+      default:
+        break;
+    }
+  };
+
+  return (
+    <EuiModal
```

Comment:
Is it easy to give the modal more width? That would give more width for the detector dropdown.

Reply:
This is definitely needed - here for a job with a `by` and `partition` field:

<img width="659" alt="image" src="https://github.com/elastic/kibana/assets/7405507/1de7ffde-40fd-4922-ba07-c0dfadec3960">

Reply:
Updated in d931543fa904ca44d10ca6e19a0855dab00062d6